.ci/ciao_down_setup.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This setup will be used to run the CI tests under ciao-down
+# under SemaphoreCI. This will avoid issues with the old packages
+# that comes in the SempahoreCI environment (Ubuntu 14.04). 
+
+set -e
+
+cidir=$(dirname "$(readlink -f "$0")")
+
+if [ -z "$LOCALCI_REPO_SLUG" ] && [ -z "$SEMAPHORE_REPO_SLUG" ]; then
+	echo "If you are running locally, please set environment variable
+		LOCALCI_REPO_SLUG for the repo you want to test. For example:
+		export LOCALCI_REPO_SLUG=clearcontainers/tests
+		If you want to test an specific PR, please also add env variable:
+		LOCALCI_PR_NUMBER, for example:
+		export LOCALCI_PR_NUMBER=100" && exit 1
+fi
+
+ciao_repo="github.com/01org/ciao"
+
+# This function will clone the repo inside the VM and
+# get the current PR that will be tested.
+function install_repo_inside_vm(){
+	export $(cat "${HOME}/ci_environment")
+	repo_slug="$LOCALCI_REPO_SLUG"
+	pr_number="$LOCALCI_PR_NUMBER"
+	if [ -n "$SEMAPHORE" ]; then
+		repo_slug="$SEMAPHORE_REPO_SLUG"
+		pr_number="$PULL_REQUEST_NUMBER"
+	fi
+	go get -d "github.com/${repo_slug}" || true
+	if [ -n "$pr_number" ]; then
+		pushd "${GOPATH}/src/github.com/${repo_slug}"
+		git fetch origin "refs/pull/${pr_number}/head:pull_${pr_number}"
+		git checkout "pull_${pr_number}"
+	fi
+}
+
+# Install ciao-down dependencies
+source /etc/os-release
+ciao_down_deps="qemu xorriso"
+if [ "$ID" == "ubuntu" ]; then
+	sudo -E apt-get update
+	eval sudo -E apt-get install -y "$ciao_down_deps"
+elif [ "$ID" == "fedora" ]; then
+	eval sudo -E dnf install -y "$ciao_down_deps"
+else
+	echo "Please install $ciao_down_deps manually and then execute this script"
+fi
+
+# Get ciao-down repository
+go get -u -t -d -v "${ciao_repo}/..." || true
+pushd "${GOPATH}/src/${ciao_repo}"
+go install -v ./...
+popd
+export PATH=$GOPATH/bin:$PATH
+
+# Provide access to /dev/kvm to the curent user to launch VMs using ciao-down
+sudo chmod a+rw /dev/kvm
+
+# Execute ciao-down
+ciao-down create -debug -cpus 2 -mem 2 "file://${cidir}/data/cc3-ubuntu-xenial.yaml"
```

Comment:
Maybe don't hardcode the number of cpus. Instead rely on `$(nproc)` from the host and use half of the resources.
Same comment for memory.